shared/fs/filepreview/view-file.native.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {globalColors, globalMargins} from '../../styles'
+import {Text} from '../../common-adapters'
+import {WebView} from 'react-native'
+import type {FilePreviewProps} from './common'
+
+const webviewUrl = url => <WebView source={{uri: url}} style={stylesWebview} />
+
+// TODO: pass here real urls from the HTTP server.
+const renderFileTypeSpecial = (props: FilePreviewProps) => {
+  switch (Types.getNameExtension(props.pathItem.name).toLowerCase()) {
```

Comment:
It might be better to base this branch on `jzila/KBFS-2843-save-share-android`, and then use `react-native-mime-types` to pull the mime type out, and then make `Constants.hasMimePrefix` modeled after this: https://github.com/keybase/client/pull/11428/files#diff-d2fb78d8c5ac488ecae7ce7d5f793167R292. That way you can make this apply to all text files (`text/*`), all images (`image/*`), all PDFs (`application/pdf`), and even `video/*`.